raster/src/main/scala/geotrellis/raster/equalization/HistogramEqualization.scala

```diff
@@ -0,0 +1,99 @@
+package geotrellis.raster.equalization
+
+import geotrellis.raster._
+import geotrellis.raster.histogram.StreamingHistogram
+
+
+/* http://www.math.uci.edu/icamp/courses/math77c/demos/hist_eq.pdf */
+object HistogramEqualization {
+
+  private class _cmp extends java.util.Comparator[(Double, Double)] {
+    // Compare the (x, cdf(x)) pairs by x
+    def compare(left: (Double, Double), right: (Double, Double)): Int = {
+      if (left._1 < right._1) -1
+      else if (left._1 > right._1) 1
+      else 0
+    }
+  }
+
+  private val cmp = new _cmp()
+
+  private def _T(cellType: CellType, cdf: Array[(Double, Double)])(x: Double): Double = {
+    val i = java.util.Arrays.binarySearch(cdf, (x, 0.0), cmp)
+    val bits = cellType.bits
+    val cdf_x =
```

Comment:
More verbose variable names or comments would make this code a lot easier to read.